src/index.html

```diff
@@ -3,82 +3,90 @@
-    <ul class="nav">
-      <li><a href="/about">About</a></li>
-      <li><a href="/contact">Contact</a></li>
-      <li><a href="/products">Products</a></li>
-    </ul>
-    <div class="header">
+    <a href="#main" class="skip-link">Skip to main content</a>
+    <nav role="navigation">
+      <ul class="nav">
+        <li><a href="/about">About</a></li>
+        <li><a href="/contact">Contact</a></li>
+        <li><a href="/products">Products</a></li>
+      </ul>
+    </nav>
+    <header class="header">
       <h1><span>a11y</span> Fails</h1>
-    </div>
-    <div class="main">
-      <h3>Product List</h3>
-      <aside class="products">
-        <div class="card">
+    </header>
+    <main class="main" role="main" id="main" tabindex="-1">
+      <h2 class="h2">Product List</h2>
+      <ul class="products" role="list">
+        <li class="card">
           <img
+            alt="White robot"
             src="https://images.unsplash.com/photo-1485827404703-89b55fcc595e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=200&h=200&q=60"
           />
-          <h2>Item 1</h2>
+          <h3 class="h2">Item 1</h3>
           <p>
             Sesame snaps gummi bears macaroon sweet chocolate cake lemon drops
             donut. Marzipan lollipop gummies chocolate bar jelly gummies icing
             oat cake.
           </p>
-          <button class="button">Buy Now</button>
-        </div>
-        <div class="card">
+          <a href="#" class="button" aria-label="Buy Item 1 Now">Buy Now</a>
+        </li>
+        <li class="card">
           <img
+            alt="Drone hovering slightly above a human hand"
             src="https://images.unsplash.com/photo-1504890001746-a9a68eda46e2?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&h=200&q=60"
           />
-          <h2>Item 2</h2>
+          <h3 class="h2">Item 2</h3>
           <p>
             Biscuit chocolate powder cheesecake powder candy canes soufflé
             lollipop. Macaroon donut danish candy canes icing.
           </p>
-          <button class="button">Buy Now</button>
-        </div>
-        <div class="card">
+          <a href="#" class="button" aria-label="Buy Item 2 Now">Buy Now</a>
+        </li>
+        <li class="card">
           <img
+            alt="Laptop on a desk"
             src="https://images.unsplash.com/photo-1496181133206-80ce9b88a853?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=200&h=200&q=60"
           />
-          <h2>Item 3</h2>
+          <h3 class="h2">Item 3</h3>
           <p>
             Caramels chocolate halvah ice cream macaroon cupcake. Chupa chups
             cotton candy lemon drops. Carrot cake brownie topping biscuit.
           </p>
-          <button class="button">Buy Now</button>
-        </div>
-      </aside>
+          <a href="#" class="button" aria-label="Buy Item 3 Now">Buy Now</a>
+        </li>
+      </ul>
 
-      <div class="form">
+      <form class="form">
```

Comment:
Probably the only additional thing I would suggest is adjusting the spacing between form fields to provide a stronger visual indication of which label belongs to which field 😊

Reply:
I agree! I added 16px of bottom spacing to each form field and I think it helps to indicate each corresponding label/field more clearly. Fixed with commit [a2ec76e](https://github.com/5t3ph/a11y-fails/pull/6/commits/a2ec76e0c4bfcff4ce6e14b1a693d1184436e897)